### PR TITLE
[FIRRTL] Update ModuleInliner to use NLATable analysis

### PR DESCRIPTION
**Pull request by prithayan**
Use NLATable analysis instead of relying on NLA breadcrumbs.

### CODE REVIEW

File: include/circt/Dialect/FIRRTL/NLATable.h [hunk 1]
```diff
@@ -49,6 +50,15 @@ class NLATable {
   /// Remove the NLA from the analysis.
   void erase(NonLocalAnchor nlaOp);
 
+  void commonNLAs(StringAttr mod1, StringAttr mod2,
```
Comment (seldridge) [inline include/circt/Dialect/FIRRTL/NLATable.h:53]: Some documentation on this would be great.

**Review by seldridge — APPROVED** [review]
LGTM

Thanks for getting more NLATable updates into more passes. 💯

File: include/circt/Dialect/FIRRTL/NLATable.h (hunk 1 above)
Comment (prithayan) [reply include/circt/Dialect/FIRRTL/NLATable.h:59]: Added the comments.

**Comment by prithayan** [comment]
Thanks @seldridge , added the comment and marked the analysis preserved.

**Pull request merged**